### DISCUSSION TITLE
Sending/Receiving logs for peppol statistics

### DISCUSSION THREAD

**Discussion by runholen** [OxalisCommunity/oxalis#665] [General]
Hi. I hope this is the right place to ask. We are a certified peppol endpoint. Unfortunately, the person that installed oxalis with AS4 has left the company, so I was a bit surprised when I just found out that we have to start sending extended statistics to peppol in February, as I had not heard of this before now.
Trying to build this statistics, I cannot seem to find logs for the incoming and outgoing AS4 messages.
For incoming, I get receipts in /opt/oxalis/var/IN
The statistic documentation seems to refer to eg eb:UserMessage but I cannot find this in the .dat files in IN.
I can in these files also only find my own CNOxxxxxx-number (servicepoint nr) in Signature.KeyInfo.X509Data, not the senders number. And outgoing traffic I cannot find any log files for at all. So where do I find a log of incoming and outgoing AS4-messages?

**Comment by bonoweb** [comment]
Hello,
i have the same problem.

official reporting documentation says: 

> For incoming Datasets (receiving), the Sending Peppol Service Provider can be deduced from the Peppol signing certificate that is part of the received AS4 message (X.509 Certificate Subject CN value)

how can i obtain this information from oxalis?

thank you!

**Comment by wartab** [comment]
Can someone help with this?

**Comment by amarianski** [comment]
I would like to know it too!

**Comment by PPotappel** [comment]
As far as I know: To get all data needed for the statistics you have to write a Java-plugin or add your own program code to the Oxalis-code.

**Comment by aaron-kumar** [comment]
@runholen, @bonoweb , @wartab, @amarianski as I mentioned sometime back to @PPotappel you can write Oxalis plugin to store data related to Peppol reporting. 
Documentation is now further enhanced, please refer : https://github.com/OxalisCommunity/oxalis/wiki/Peppol-Reporting 

Converting to discussion as it is Not bug
